--- a/simpleddp/classes/ddpReactiveCollection.d.ts
+++ b/simpleddp/classes/ddpReactiveCollection.d.ts
@@ -116,7 +116,7 @@
      * @param {Function} f - Function which recieves new collection at each change.
      * @public
      */
-    public onChange(f: Function): ddpOnChange;
+    public onChange(f: (collection: T[]) => void): ddpOnChange;
     /**
      * Maps reactive local collection to another reactive array.
      * Specified function form {@link https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/map}.
